### PR TITLE
Provider sign in

### PR DESCRIPTION
**Pull request by UrosLITS**
Added google and Facebook sign in as well as email verification for basic sign in

### CODE REVIEW

File: lib/data/firebase_auth/firebase_auth_singleton.dart [hunk 1]
```diff
@@ -79,12 +83,103 @@ class FirebaseAuthSingleton {
...
+
+    final OAuthCredential facebookAuthCredential =
+        FacebookAuthProvider.credential(loginResult.accessToken!.tokenString);
+
+    return await _auth.signInWithCredential(facebookAuthCredential);
+  }
+
+  Future<UserCredential?> signInWithGoogle() async {
+    final GoogleSignIn _googleSignIn;
+    if (Platform.isAndroid) {
```
Comment (petarLits) [inline lib/data/firebase_auth/firebase_auth_singleton.dart]: if platform is ios send clientId, else....

File: lib/data/firebase_auth/firebase_auth_singleton.dart [hunk 2]
```diff
@@ -79,12 +83,103 @@ class FirebaseAuthSingleton {
...
+      _googleSignIn = await GoogleSignIn(scopes: ['email']);
+    } else {
+      _googleSignIn = await GoogleSignIn(
+          clientId:
+              '576996108646-d4gch7ds48igle7gvhi224qhgfmqfslg.apps.googleusercontent.com',
```
Comment (petarLits) [inline lib/data/firebase_auth/firebase_auth_singleton.dart]: use constants

File: lib/data/firebase_auth/firebase_auth_singleton.dart [hunk 3]
```diff
@@ -79,12 +83,103 @@ class FirebaseAuthSingleton {
...
+    final GoogleSignInAuthentication gAuth = await gUser.authentication;
+
+    final OAuthCredential credential = GoogleAuthProvider.credential(
+      accessToken: gAuth.accessToken,
+      idToken: gAuth.idToken,
+    );
+
+    final result = await _auth.signInWithCredential(credential);
+
+    if (credential.providerId == 'google.com') {
```
Comment (petarLits) [inline lib/data/firebase_auth/firebase_auth_singleton.dart]: use constants

**Review by petarLits — CHANGES_REQUESTED** [review]

File: lib/data/firebase_auth/firebase_auth_singleton.dart [hunk 4]
```diff
@@ -79,12 +83,101 @@ class FirebaseAuthSingleton {
...
+        onTimeout: () {
+      throw ServerConnectionException();
+    });
+  }
+
+  Future<void> checkEmailVerified() async {
+    User? user = _auth.currentUser;
+
+    if (user!.emailVerified) {
+      print(emailVerified);
```
Comment (milicaLits) [inline lib/data/firebase_auth/firebase_auth_singleton.dart]: unecessary print

File: lib/presentation/register/view/create_provider_account.dart [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+      );
+    });
+  }
+
+  Future<bool> onBackPressed(BuildContext context) async {
+    if (name != null ||
+        lastName != null ||
+        password != null ||
+        confirmPassword != null) {
+      final result = await showDialog(
```
Comment (milicaLits) [inline lib/presentation/register/view/create_provider_account.dart:218]: you could use a getter instead of having all these here in condition

**Review by milicaLits — CHANGES_REQUESTED** [review]

**Review by milicaLits — APPROVED** [review]

**Review by petarLits — APPROVED** [review]

**Pull request merged**